tests/unit/NetworkTest.js

```diff
@@ -209,3 +221,45 @@ test('consecutive API calls eventually succeed when authToken is expired', () =>
             expect(chatList).toEqual(TEST_CHAT_LIST);
         });
 });
+
+test('retry network request if auth and credentials are not read from Onyx yet', () => {
+    Onyx.addDelayToConnectCallback(3000);
+
+    // Reset isReady for this unit test, because Onyx.clear() in beforeEach resolves the
+    // Onyx.connect callbacks with null values
+    Network.setIsReady(false);
+
+    // Set values to trigger Onyx.connect callbacks
+    Onyx.merge(ONYXKEYS.CREDENTIALS, {});
+    Onyx.merge(ONYXKEYS.SESSION, {});
+
+    // Functions to mock and that we want to track
+    const spyNetworkSetIsReady = jest.spyOn(Network, 'setIsReady');
+    const spyHttpUtilsXhr = jest.spyOn(HttpUtils, 'xhr').mockImplementation(() => Promise.resolve({}));
+
+    // Given a test user login and account ID
+    const TEST_USER_LOGIN = 'test@testguy.com';
+    fetchAccountDetails(TEST_USER_LOGIN);
```

Comment:
Why do we need to `fetchAccountDetails()`? Are we just making some arbitrary request that requires an `authToken`?

Reply:
Yeap, exactly, it's just an arbitrary request which can be retried if the auth token + credentials are not read from storage yet.

Reply:
That would be a good thing to note maybe in a comment.